src/Soil-Core/SOTransaction.class.st

```diff
@@ -45,32 +45,34 @@ SOTransaction >> atObjectId: objectId putObject: anObject [
 { #category : #public }
 SOTransaction >> behaviorDescriptionFor: aClass [
 	| classDescription id |
-	classDescriptions 
-		at: aClass soilBehaviorIdentifier 
+	classDescriptions
+		at: aClass soilBehaviorIdentifier
 		ifPresent: [ :description | ^ description ].
-	soil behaviorRegistry 
-		nameAt: aClass soilBehaviorIdentifier 
+	soil behaviorRegistry
+		nameAt: aClass soilBehaviorIdentifier
 		ifPresent: [ :oid | | desc |
-			desc := (oid index = 2) 
-				ifTrue: [(SOBehaviorDescription for: SOBehaviorDescription) objectId: oid ]
-				ifFalse: [ (self objectWithId: oid ifNone: [Error signal ]) objectId: oid; yourself ].
+			desc := (oid index = 2)
+				ifTrue: [SOBehaviorDescription meta objectId: oid ]
+				ifFalse: [ | foundDesc |
+					 "the description in the database might not be current, if not, we create a new one with the same ID"
+					 foundDesc := (self objectWithId: oid ifNone: [Error signal ]) objectId: oid; yourself.
+					 foundDesc isCurrent ifFalse: [ (SOBehaviorDescription for: aClass) objectId: oid ] ifTrue: [foundDesc]].
```

Comment:
Is it necessary to put an objectId: there? The current concept includes having different objectIds for different behavior versions. So this looks a bit cheesy

Reply:
yes, this is odd... I will look.